### PR TITLE
Define API responses correctly instead default 200

### PR DESCRIPTION
**Pull request by doncem**
### Change description ###

Swagger uses default 200 for all endpoints unless specified otherwise with spring's `@ResponseStatus`

before:

screen shot 2018-03-02 at 09 05 54

after:

screen shot 2018-03-02 at 09 29 03

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sendletter/controllers/ReportController.java [hunk 1]
```diff
@@ -34,6 +38,10 @@ public ReportController(
 
     @PostMapping(path = "/print-status-check")
     @ApiOperation(value = "Execute action for print state")
+    @ApiResponses({
+        @ApiResponse(code = 401, message = "Invalid service authorisation")
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/sendletter/controllers/ReportController.java]: 401 is missing auth header. invalid is 403

Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/sendletter/controllers/ReportController.java]: There's also validation whether service has access to perform action and response is the same - 401 😕 . Made assumption it's "invalid service auth" for both checks

Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/sendletter/controllers/ReportController.java]: sooo.. `401 Unauthorised` and `403 Forbidden`. the for both assertions we use 401 and supply error message. initial post/get endpoints we have has `Invalid service authorisation header` as a message. but they don't check whether service is allowed to perform action so makes sense to include header.

tl;dr what message do you think it should be?

File: src/main/java/uk/gov/hmcts/reform/sendletter/controllers/SendLetterController.java [hunk 2]
```diff
@@ -114,6 +125,10 @@ public SendLetterController(
 
     @PutMapping(path = "/{id}/is-failed")
     @ApiOperation(value = "Update failed status when letter was sent to dead letter queue")
+    @ApiResponses({
+        @ApiResponse(code = 401, message = "Invalid service authorisation")
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/sendletter/controllers/SendLetterController.java]: SonarQube may complain about repeated string

Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/reform/sendletter/controllers/SendLetterController.java]: Good point.It will be good if we could run our PR against local SONAR instance so issues can be identified earlier in the CD process.

Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/sendletter/controllers/SendLetterController.java]: Didn't I see Sonarqube check somewhere in PRs?

Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/reform/sendletter/controllers/SendLetterController.java]: I meant running `./gradlew sonar` locally :) if that makes sense.Of course you need a local instance of SONAR running(which is just docker container)
